### ISSUE TITLE
promote_scalars_to_symbols breaks in 0.13.2

### ISSUE THREAD

**Issue by leporia** [spcl/dace#1033]
**Describe the bug**
In version 0.13.2 when promote_scalars_to_symbols is applied it substitutes indices with wrong values. It only happens with array of more than 1 dimension. All indices after the first are substituted with `0`.

**To Reproduce**
For some reason I wasn't able to reproduce the bug with an handcrafted SDFG. Here is an auto generated one that shows the bug.
[untransformed.zip](https://github.com/spcl/dace/files/8974300/untransformed.zip)

And here is the transformation code
```
import dace
from dace.sdfg import SDFG

sdfg = SDFG("test").from_file("untransformed.sdfg")

from dace.transformation.passes import scalar_to_symbol as scal2sym

for sd in sdfg.all_sdfgs_recursive():
        promoted = scal2sym.promote_scalars_to_symbols(sd)

sdfg.save("transformed.sdfg")
```
On `_state16_17` before the transformation the array `A` was modified at indices `size_1-3` and `size_2-5`. But after the transformation the second index is substituted by `0` directly in the edge.
It seems that on every 2 dimensional array it shows the same behavior.

Because I wasn't able to reproduce the bug with an handcrafted SDFG it is possible that the bug is caused by abusing some syntax.

**Expected behavior**
The index is substituted correctly with more than 1 dimensional arrays.

**Desktop (please complete the following information):**
- OS: Fedora Linux 35 (Workstation Edition)
- DaCe version 0.13.2

**Comment by tbennun** [comment]
When was the last time it worked?

**Comment by leporia** [comment]
With version 0.13.1 of DaCe it works. I'll try with Python tasklets if you say that those are more reliable.

**Comment by tbennun** [comment]
I found and fixed the issue in #1032. Thanks again for reporting!

**Comment by leporia** [comment]
I confirm. Now it seems to be fixed
